src/ape_ethereum/multicall/handlers.py

```diff
@@ -0,0 +1,262 @@
+        """
+        Perform the Multicall call. This call will trigger again every time the ``Call`` object
+        is called.
+
+        Raises:
+            :class:`UnsupportedChain`: If there is not an instance of Multicall3 deployed
+              on the current chain at the expected address.
+
+        Args:
+            **call_kwargs: the kwargs to pass through to the call handler.
+
+        Returns:
+            Iterator[Any]: the sequence of values produced by performing each call stored
+              by this instance.
+        """
+        self._result = self.handler(self.calls, **call_kwargs)  # type: ignore [operator]
+        return self._decode_results()
+
+    def as_transaction(self, **txn_kwargs) -> TransactionAPI:
+        """
+        Encode the Multicall transaction as a ``TransactionAPI`` object, but do not execute it.
+
+        Returns:
+            :class:`~ape.api.transactions.TransactionAPI`
+        """
+        return self.handler.as_transaction(self.calls, **txn_kwargs)  # type: ignore[attr-defined]
+
+
+class Transaction(BaseMulticall):
+    """
+    Create a sequence of calls to execute at once using `eth_sendTransaction`
+    via the Multicall3 contract.
+
+    Usage example::
+
+        from ape_ethereum import multicall
+
+        txn = multitxn.Transaction()
+        txn.add(contract.myMethod, *call_args)
+        txn.add(contract.myMethod, *call_args)
+        ...  # Add as many calls as desired to execute
+        txn.add(contract.myMethod, *call_args)
+        a, b, ..., z = txn(sender=my_signer)  # Sends the multical transaction
+    """
+
+    def __init__(self) -> None:
+        super().__init__(ContractTransactionHandler)
+
+    def _validate_calls(self, **txn_kwargs) -> None:
+        if self._handler_method_abi == AGGREGATE3VALUE_METHOD:
+            required_value = sum(call[2] for call in self.calls)  # type: ignore[misc]
+            if "value" not in txn_kwargs:
+                raise ValueRequired(required_value)
+
+            value = self.conversion_manager.convert(txn_kwargs["value"], int)
+
+            if required_value < value:
+                raise ValueRequired(required_value)
+
+        # NOTE: Won't fail if `value` is provided otherwise (won't do anything either)
+
+    def __call__(self, **txn_kwargs) -> ReceiptAPI:
+        """
+        Execute the Multicall transaction. The transaction will broadcast again every time
+        the ``Transaction`` object is called.
+
+        Raises:
+            :class:`UnsupportedChain`: If there is not an instance of Multicall3 deployed
+              on the current chain at the expected address.
+
+        Args:
+            **txn_kwargs: the kwargs to pass through to the transaction handler.
+
+        Returns:
+            :class:`~ape.api.transactions.ReceiptAPI`
+        """
+        self._validate_calls(**txn_kwargs)
+        return self.handler(self.calls, **txn_kwargs)  # type: ignore [operator]
```

Comment:
```suggestion
        return self.handler(self.calls, **txn_kwargs)  # type: ignore[operator]
```

Although, is this type ignore actually needed?

Reply:
Yes